x/incentives/keeper/distribute.go

```diff
@@ -344,32 +375,78 @@ func (k Keeper) Distribute(ctx sdk.Context, gauges []types.Gauge) (sdk.Coins, er
 
 	locksByDenomCache := make(map[string][]lockuptypes.PeriodLock)
 	totalDistributedCoins := sdk.Coins{}
+	var gaugeDistributedCoins, gaugeDistributedCoinsCL sdk.Coins
 	for _, gauge := range gauges {
-		filteredLocks := k.getDistributeToBaseLocks(ctx, gauge, locksByDenomCache)
-		// send based on synthetic lockup coins if it's distributing to synthetic lockups
-		var gaugeDistributedCoins sdk.Coins
-		var err error
-		if lockuptypes.IsSyntheticDenom(gauge.DistributeTo.Denom) {
-			gaugeDistributedCoins, err = k.distributeSyntheticInternal(ctx, gauge, filteredLocks, &distrInfo)
+		// get pool Id corresponding to the gaugeId
+		incParams := k.GetParams(ctx).DistrEpochIdentifier
+		currEpoch := k.ek.GetEpochInfo(ctx, incParams)
+
+		isCLPoolGauge, pool := k.IsPoolValidAndIsCLPoolGauge(ctx, gauge.Id, currEpoch.Duration)
+		// only want to run this logic if the gaugeId is associated with CL PoolId
+		if isCLPoolGauge {
+			// since we're only doing osmo incentive we can assume there to be only 1 coin therefore coins[0]
+			incentiveRecord := cltypes.IncentiveRecord{
+				PoolId:           pool.GetId(),
+				IncentiveDenom:   gauge.Coins[0].Denom,
+				IncentiveCreator: k.ak.GetModuleAddress(types.ModuleName),
+				RemainingAmount:  sdk.Dec(gauge.Coins[0].Amount),
+				// calculate amount of tokens to emit per second
+				// for ex: 10000tokens to be distributed over 1day epoch will be 1000 tokens ÷ 86,400 seconds ≈ 0.0116 tokens per second
+				EmissionRate: sdk.NewDecFromInt(gauge.Coins[0].Amount).Quo(sdk.NewDecFromInt(sdk.NewInt(int64(currEpoch.Duration.Seconds())))),
```

Comment:
Discussed offline to convert `sdk.NewDecFromInt(sdk.NewInt(` -> `sdk.NewDec(` and `.Quo` -> `.QuoTruncate`